internal/fs/inode/dir.go

```diff
@@ -366,10 +365,13 @@ func findExplicitFolder(ctx context.Context, bucket *gcsx.SyncerBucket, name Nam
 		return nil, fmt.Errorf("error in get folder for lookup : %w", folderErr)
 	}
 
+	folderObject := folderResult.ConvertFolderToMinObject()
+	folderObject.Name = name.objectName
```

Comment:
When would the folderName and the requestedName be different and would it be possible to move it inside the convertFolderToMinObject() if its needed in more places than here.

Reply:
This is the test representing TestCopyDirectoryInNonExistingDirectory case when folderName and the requestedName are different.

Reply:
Moved name inside and also added test case for same.